Caboodle/Types/Location.shared.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+using static System.Math;
+
+namespace Microsoft.Caboodle
+{
+
+
+	public enum DistanceUnits
+	{
+		Kilometers,
+		Miles
+	}
+
+
+	public class Location
+	{
+		public Location()
+		{
+
+		}
+		public Location(double latitude, double longitude)
+		{
+			Latitude = latitude;
+			Longitude = longitude;
+			TimestampUtc = DateTimeOffset.UtcNow;
+		}
+		public Location(double latitude, double longitude, DateTimeOffset timestamp)
+		{
+			Latitude = latitude;
+			Longitude = longitude;
+			TimestampUtc = timestamp;
+		}
+		public Location(Location point)
+		{
+			if (point == null)
+				throw new ArgumentNullException(nameof(point));
+
+			Latitude = point.Latitude;
+			Longitude = point.Longitude;
+			TimestampUtc = DateTimeOffset.UtcNow;
+		}
+
+		public DateTimeOffset TimestampUtc { get; set; }
+
+		public double Latitude { get; set; }
+
+		public double Longitude { get; set; }
+
+
+		public static double CalculateDistance(Location locationStart, Location locationEnd, DistanceUnits units = DistanceUnits.Miles) =>
+			CalculateDistance(locationStart.Latitude, locationStart.Longitude, locationEnd.Latitude, locationEnd.Longitude, units);
+
+		public static double CalculateDistance(
+			double latitudeStart, double latitudeEnd,
+			double longitudeStart, double longitudeEnd, DistanceUnits units = DistanceUnits.Miles)
```

Comment:
I recall a discussion about default parameters. They are nice to use but may bite us later. Maybe use overloads instead.

Reply:
no worries, yeah can remove. I am cools